flink-tests/src/test/java/org/apache/flink/test/recovery/ProcessFailureCancelingITCase.java

```diff
@@ -269,57 +231,20 @@ public Long map(Long value) throws Exception {
-     * @return {@link DispatcherGateway}
-     * @throws Exception if something goes wrong
-     */
-    static DispatcherGateway retrieveDispatcherGateway(
-            RpcService rpcService, HighAvailabilityServices haServices) throws Exception {
-        final LeaderConnectionInfo leaderConnectionInfo =
-                LeaderRetrievalUtils.retrieveLeaderConnectionInfo(
-                        haServices.getDispatcherLeaderRetriever(), Duration.ofSeconds(10L));
-
-        return rpcService
-                .connect(
-                        leaderConnectionInfo.getAddress(),
-                        DispatcherId.fromUuid(leaderConnectionInfo.getLeaderSessionId()),
-                        DispatcherGateway.class)
-                .get();
-    }
-
-    private void waitUntilAllSlotsAreUsed(DispatcherGateway dispatcherGateway, Time timeout)
-            throws ExecutionException, InterruptedException {
-        FutureUtils.retrySuccessfulWithDelay(
-                        () -> dispatcherGateway.requestClusterOverview(timeout),
-                        Time.milliseconds(50L),
-                        Deadline.fromNow(Duration.ofMillis(timeout.toMilliseconds())),
-                        clusterOverview ->
-                                clusterOverview.getNumTaskManagersConnected() >= 1
-                                        && clusterOverview.getNumSlotsAvailable() == 0
-                                        && clusterOverview.getNumSlotsTotal() == 2,
-                        TestingUtils.defaultScheduledExecutor())
-                .get();
-    }
-
-    private Collection<JobID> waitForRunningJobs(ClusterClient<?> clusterClient, Time timeout)
-            throws ExecutionException, InterruptedException {
-        return FutureUtils.retrySuccessfulWithDelay(
-                        CheckedSupplier.unchecked(clusterClient::listJobs),
-                        Time.milliseconds(50L),
-                        Deadline.fromNow(Duration.ofMillis(timeout.toMilliseconds())),
-                        jobs -> !jobs.isEmpty(),
-                        TestingUtils.defaultScheduledExecutor())
-                .get().stream()
-                .map(JobStatusMessage::getJobId)
-                .collect(Collectors.toList());
+    private static void waitUntilAtLeastOneTaskHasBeenDeployed(TestProcess taskManagerProcess)
+            throws InterruptedException, TimeoutException {
+        CommonTestUtils.waitUtil(
+                () ->
+                        taskManagerProcess
+                                .getProcessOutput()
+                                .toString()
+                                .contains(TASK_DEPLOYED_MARKER),
+                Duration.ofMinutes(2),
+                null);
     }
 
-    private void printProcessLog(String processName, String log) {
-        if (log == null || log.length() == 0) {
+    private void printOutput(String processName, String logContents) {
```

Comment:
The parameter renaming didn't work properly: `logContents` is used when actually printing content